client/src/pages/ProfileDetails/ProfileDetails.tsx

```diff
@@ -0,0 +1,80 @@
+import useStyles from './useStyles';
+import { Box, Typography, Card, CardMedia, CardContent } from '@material-ui/core';
+import RoomIcon from '@material-ui/icons/Room';
+import BookingForm from './BookingForm/BookingForm';
+import fetchProfile from '../../helpers/APICalls/fetchProfile';
+import { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { Profile } from '../../interface/Profile';
+
+const ProfileDetails = (): JSX.Element => {
+  const classes = useStyles();
+  const [profile, setProfile] = useState<Profile>({
+    firstName: '',
+    lastName: '',
+    description: '',
+    address: '',
+    phoneNumber: '',
+    dateOfBirth: new Date('December 17, 1995 03:24:00'),
+    available: false,
+    availability: '',
+    gender: '',
+    email: '',
+  });
+
+  // const { id }: { id: string } = useParams();
```

Comment:
Please remove commented code or add a TODO comment

Reply:
I'm having an issue with the useParams hook not working. It was previously working but now is unable to get the id parameter from the URL. Could you take a look at this?

Reply:
@jb162 Are we getting any error from the above line of code? Also, did we change the profile details route?

Reply:
From Line 25, I am getting "Unhandled Rejection (TypeError): Cannot read properties of undefined (reading 'profile')". I tried to log the parameters but the useParams hook is not working at all. When logging the params variable that is set equal to useParams(), I receive an object but it does not include the id parameter. The profile details route has not been changed and is set to "/profile-details/:id".

Reply:
@jb162 The useParams hook is generic, we need to tell typescript which params we are using by specifying the value of the generic

```
const { sumParams } = useParams<{ id?: string }>();
```

Reply:
I was still having this issue even after implementing the above line. For some reason, useParams() returned an empty object every time. The only way I was able to fix it was by switching from a ProtectedRoute to a normal Route component.